internal/daemon/daemon.go

```diff
@@ -853,7 +853,7 @@ func (d *Daemon) addExtensionServers(preInit bool, fallbackCert *shared.CertInfo
 		} else {
 			// Generate a dedicated certificate or load the custom one if it exists.
 			// When updating the additional listeners the dedicated certificate from before will be reused.
-			cert, err = shared.KeyPairAndCA(d.os.CertificatesDir, serverName, shared.CertServer, true)
+			cert, err = shared.KeyPairAndCA(d.os.CertificatesDir, serverName, shared.CertServer, shared.CertOptions{AddHosts: true})
```

Comment:
Well the cluster cert needs to be identical across all nodes so we don't gain anything by assigning it the name of the first node. 

As for extension servers, we shouldn't be changing the cert they give us.

Reply:
Yeah I guess (1) we can leave like it is. It doesn't hurt having the name of the first node in the cert.

Regarding (2) I was just thinking it could make sense to use the additional server name in the SAN, otherwise we also just set the host name of the first node. But at least it would be consistent with the `cluster.crt`.

Reply:
Jup if you specify `DedicatedCertificate: true` then we neither use the `cluster.crt` nor a custom one but instead create a fresh one as the user requested a dedicated one.